node_modules/oae-content/lib/api.js

```diff
@@ -482,60 +483,73 @@ var _createContent = function(ctx, contentId, revisionId, resourceSubType, displ
         return callback(validator.getFirstError());
     }
 
-    // We need to check if we can interact with the additional members
-    // before we create the content item
-    var additionalMemberIds = _.keys(additionalMembers);
-    PrincipalsDAO.getPrincipals(additionalMemberIds, null, function(err, principals) {
+    // Get the principal ids, creating guest users if necessary    
+    PrincipalsAPI.getOrCreateGuestUsers(ctx, _.keys(additionalMembers), function(err, principalIds) {
         if (err) {
             return callback(err);
-        } else if (_.keys(principals).length !== additionalMemberIds.length) {
-            return callback({'code': 400, 'msg': 'One or more target members being granted access do not exist'});
         }
 
-        principals = _.values(principals);
+        var permissions = {};
+        _.each(additionalMembers, function(role, principal) {
+            permissions[principalIds[principal]] = role;
+        });
+
+        var additionalMemberIds = _.values(principalIds);
 
-        AuthzAPI.canInteract(ctx, ctx.tenant().alias, principals, function(err, canInteract, illegalPrincipalIds) {
+        // We need to check if we can interact with the additional members
+        // before we create the content item
+        PrincipalsDAO.getPrincipals(additionalMemberIds, null, function(err, principals) {
```

Comment:
Do we need to do both `PrincipalsAPI.getOrCreateGuestUsers` and `PrincipalsDAO.getPrincipals` ? Can they be combined in a single function?